### PR TITLE
fix: validate sobjects related folders

### PR DESCRIPTION
**Pull request by CristiCanizales**
- Adds logging in each step and right validation for presence of folders for sObjectsDefinitions.e2e.ts file

@W-16825635@

[Test run](https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/11022085552) ✅

### CODE REVIEW

File: test/specs/sObjectsDefinitions.e2e.ts [hunk 1]
```diff
@@ -23,6 +23,9 @@ describe('SObjects Definitions', async () => {
   });
 
   step('Check Custom Objects Customer and Product are within objects folder', async () => {
+    utilities.log(
+      `${testSetup.testSuiteSuffixName} - Check Custom Objects Customer and Product are within objects folder`
```
Comment (diyer) [inline test/specs/sObjectsDefinitions.e2e.ts:27]: nit: can we change this to "Check Custom Objects 'Customer__c' and 'Product__c' are within objects folder"

Comment (diyer) [reply test/specs/sObjectsDefinitions.e2e.ts:27]: nm, looks like this is an existing step

File: test/specs/sObjectsDefinitions.e2e.ts [hunk 2]
```diff
@@ -86,6 +90,9 @@ describe('SObjects Definitions', async () => {
   });
 
   step('Refresh SObject Definitions for Custom SObjects', async () => {
```
Comment (diyer) [inline test/specs/sObjectsDefinitions.e2e.ts:92]: not related to this PR, but looks like we can refactor out the step method in this test as it seems to be doing the same thing.

File: test/specs/sObjectsDefinitions.e2e.ts [hunk 3]
```diff
@@ -203,11 +214,34 @@ describe('SObjects Definitions', async () => {
...
+    sobjectsTreeItem = sobjectsTreeItem!;
+
+    await sobjectsTreeItem.expand();
+    await expect(await sobjectsTreeItem.isExpanded()).toBe(true);
+    await utilities.pause(utilities.Duration.seconds(1));
+
+    // Verify if 'standardObjects' folder is within 'sobjects'
+    let standardObjectsTreeItem = await utilities.getVisibleChild(sfdxTreeItem, 'standardObjects');
     await expect(standardObjectsTreeItem).not.toEqual(undefined);
+    standardObjectsTreeItem = standardObjectsTreeItem!;
```
Comment (peternhale) [inline test/specs/sObjectsDefinitions.e2e.ts:243]: I understand that the expect does not provide TS with enough informaion to continue without applying the `!`, but use of `!` is discouraged. 
```suggestion
    if (!standardObjectsTreeItem) {
      fail('Could not locate standardObjects in tree view');
    }
    ...
```

File: test/specs/sObjectsDefinitions.e2e.ts (hunk 2 above)
Comment (peternhale) [reply test/specs/sObjectsDefinitions.e2e.ts:93]: There are three paths to check All, Standard and Custom and given we do not have unit tests how else do we confirm that the command is setup correctly?

**Review by peternhale — CHANGES_REQUESTED** [review]
Just a nit on the use of `!` operator. This test has a ton of duplicate code. Would you have a go at that as part of this PR?

**Review by peternhale — APPROVED** [review]

**Pull request merged**